server/server.go

```diff
@@ -0,0 +1,35 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// Middleware is a function that decorates an http.Handler
+type Middleware func(handler http.Handler) http.Handler
+
+// ApplyMiddleware creates a final http Handler with a slice of middleware functions applied
+func ApplyMiddleware(handlers []Middleware, final http.Handler) http.Handler {
+	for _, handler := range handlers {
+		final = handler(final)
+	}
+	return final
+}
+
+// ServeMux serves a set of endpoints defined in a ServerMux with the defined middleware applied
+func ServeMux(httpd *http.ServeMux, mw []Middleware, servicePort string) *http.Server {
```

Comment:
Curious, is there a reason we're implementing this ourselves instead of using something like gorilla mux?